src/canister/wallet.rs

```diff
@@ -27,8 +28,18 @@ fn get_wallet_principal<'a>(
     account_name: impl AsRef<str>,
     wallet_id_path: impl Into<Option<&'a str>>,
 ) -> Result<Principal> {
-    let wallet_id_path = wallet_id_path.into().unwrap_or(WALLET_IDS_PATH);
-    let json_str = read_to_string(wallet_id_path)?;
+    let wallet_id_path = PathBuf::from(wallet_id_path.into().unwrap_or(WALLET_IDS_PATH));
```

Comment:
```suggestion
    let path_str = wallet_id_path.into().unwrap_or(WALLET_IDS_PATH);
    let wallet_id_path = PathBuf::from(path_str);
```
Looks more readable for me.